libpromises/evalfunction.c

```diff
@@ -251,7 +264,8 @@ static JsonElement* VarRefValueToJson(EvalContext *ctx, const FnCall *fp, const
                     }
                     else if (var->ref->num_indices - index_offset > 1)
                     {
-                        Log(LOG_LEVEL_DEBUG, "%s: got ref with starting depth %zd and index count %zd", fp->name, index_offset, var->ref->num_indices);
+                        Log(LOG_LEVEL_DEBUG, "%s: got ref with starting depth %zd and index count %zd",
+                            fp ? fp->name : "VarRefValueToJson", index_offset, var->ref->num_indices);
```

Comment:
Can you make this `fp` check at the top of the function instead, and store the name in a different variable. Then you don't need to check this every place where it's used.

Same goes for several uses below this one.

Alternatively, since `fp` is only used to fetch the name AFAICS, perhaps you shouldn't pass in a null pointer at all, but you should change it to a string and pass in `"LookupVarRefToJson"` from the one function that now passes null. It still won't be fully verbose in the sense that the real calling function from the policy is masked, but it would take quite a bit of data passing to get that information all the way here.